csharp/src/Drivers/Apache/Thrift/BitmapUtilities.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace Apache.Arrow.Adbc.Drivers.Apache.Thrift
+{
+    internal static class BitmapUtilities
+    {
+        /// <summary>
+        /// Gets the "validity" bitmap buffer from a 'nulls' bitmap.
+        /// </summary>
+        /// <param name="nulls">The bitmap of rows where the value is a null value (i.e., "invalid")</param>
+        /// <param name="nullCount">Returns the number of bits set in the bitmap.</param>
+        /// <returns>A <see cref="ArrowBuffer"/> bitmap of "valid" rows (i.e., not null values).</returns>
+        /// <remarks>Inverts the bits in the incoming bitmap to reverse the null to valid indicators.</remarks>
+        internal static ArrowBuffer GetValidityBitmapBuffer(byte[] nulls, out int nullCount)
+        {
+            byte[] valids = new byte[nulls.Length];
+            for (int i = 0; i < nulls.Length; i++)
+            {
+                valids[i] = (byte)~nulls[i];
+            }
+            nullCount = BitUtility.CountBits(nulls);
```

Comment:
This seems problematic for the case where the number of values is not an exact multiple of eight.

Reply:
@CurtHagenlocher - Updated to switch all bits, one-by-one. Not confident about byte ordering coming from Spark server.